### PR TITLE
[Feature] Making Synetune Optimizers compatible with trainsize fidelity

### PR DESCRIPTION
**Pull request by timruhkopf**
We  need to convert the budget to an integer valued one.

### CODE REVIEW

File: carps/optimizers/synetune.py [hunk 1]
```diff
@@ -154,7 +155,8 @@ def convert_configspace(self, configspace: ConfigurationSpace) -> dict[str, Any]
         for k, v in configspace.items():
             configspace_st[k] = configspaceHP2syneTuneHP(v)
         if self.fidelity_enabled:
-            configspace_st[self.fidelity_type] = self.max_budget
+            configspace_st[self.fidelity_type] = self.max_budget if not self.convert else int(
+                1000 * self.max_budget)
```
Comment (benjamc) [inline carps/optimizers/synetune.py:159]: no magic numbers, make this a class variable

File: carps/optimizers/synetune.py [hunk 2]
```diff
@@ -216,7 +221,8 @@ def convert_to_synetrial(self, trial_info: TrialInfo) -> SyneTrial:
         """
         syne_config = dict(trial_info.config)
         if self.fidelity_enabled:
-            syne_config[self.fidelity_type] = trial_info.budget
+            syne_config[self.fidelity_type] = trial_info.budget if not self.convert else int(
+                1000 * trial_info.budget)
```
Comment (benjamc) [inline carps/optimizers/synetune.py:225]: same here, make variable/constant

File: carps/optimizers/synetune.py [hunk 3]
```diff
@@ -246,7 +252,9 @@ def tell(self, trial_info: TrialInfo, trial_value: TrialValue):
             experiment_result = {self.task.objectives[i]: cost[i] for i in range(len(cost))}
 
         if self.optimizer_name == "MOASHA":
```
Comment (benjamc) [inline carps/optimizers/synetune.py:254]: do not check for MOASHA but if it is MF
`if self.task.is_multifidelity`

File: carps/optimizers/synetune.py [hunk 4]
```diff
@@ -304,15 +312,33 @@ def _setup_optimizer(self) -> SyneTrialScheduler:
...
         if self.optimizer_name in mf_optimizer_dicts["with_mf"]:
             _optimizer_kwargs["resource_attr"] = self.fidelity_type
             # _optimizer_kwargs["max_t"] = self.max_budget  # TODO check how to set n trials / wallclock limit for synetune
 
+            # for floating point resources like trainsize, we need to convert them to integer for synetune
+            if "grace_period" in self.optimizer_kwargs and isinstance(
+                    self.optimizer_kwargs["grace_period"], float):
+                self.convert = True
+                _optimizer_kwargs["grace_period"] = int(
+                    1000 * self.optimizer_kwargs["grace_period"])
```
Comment (benjamc) [inline carps/optimizers/synetune.py:330]: same here, no magic numbers

**Review by benjamc — CHANGES_REQUESTED** [review]

File: carps/optimizers/synetune.py (hunk 3 above)
Comment (timruhkopf) [reply carps/optimizers/synetune.py:262]: i am actually not sure if this is only MOASHA specific

Comment (benjamc) [reply carps/optimizers/synetune.py:262]: thought it would handle all MF stuff

**Review by benjamc — APPROVED** [review]

**Pull request merged**